routes/login/login.controller.js

```diff
@@ -180,24 +188,19 @@ const postLoginCode = async (req, res, next) => {
     })
   }
 
-  let user
-
-  if (process.env.CTBS_SERVICE_URL && req.body.code) {
-    user = await request({
-      method: 'GET',
-      uri: `${process.env.CTBS_SERVICE_URL}/${req.body.code}`,
-      json: true,
-    })
-  } else {
-    user = API.getUser(req.body.code || null)
-  }
+  // check if code is valid
+  let validCode = DB.validateCode(req.body.code)
 
-  if (!user) {
+  if (!validCode) {
     throw new Error(`[POST ${req.path}] user not found for access code "${req.body.code}"`)
   }
 
+  // populate the session with empty variables in the format of user.json
   // setting req.session = {obj} causes an error, so assign the keys one at a time
-  Object.keys(user).map(key => (req.session[key] = user[key]))
+  Object.keys(user).map(key => (req.session[key] = {}))
```

Comment:
I think a req.session.db *could* work. The reason I took this approach is that there are issues where, for example, the sin page on its post call was already expecting the sin to be in the session, which I don't think it would be in the new flow. Initially we're just validating the Access code, and not pulling the rest of the info out of the DB until the DoB call, right?

I guess, in my head, the flow was - check if access code exists, add it to session. Add sin and DoB to session as they are input, validate them in DoB post, and retrieve the rest of the data. (We could re-write the SIN / DoB from that call as well if needed - addressing one of your other comments)